dso-l2/src/main/java/com/tc/services/PlatformServiceProvider.java

```diff
@@ -0,0 +1,38 @@
+package com.tc.services;
+
+import com.tc.objectserver.api.ManagedEntity;
+import com.tc.objectserver.impl.DistributedObjectServer;
+import org.terracotta.entity.ServiceConfiguration;
+import org.terracotta.entity.ServiceProviderCleanupException;
+import org.terracotta.monitoring.PlatformService;
+
+import java.util.Collection;
+import java.util.Collections;
+
+/**
+ * @author vmad
+ */
+public class PlatformServiceProvider implements BuiltInServiceProvider {
+
+    private final PlatformServiceImpl platformService;
+
+    public PlatformServiceProvider(DistributedObjectServer.DumpHelper dumpHelper) {
```

Comment:
This wouldn't need to know about so many things if you just passed in a PlatformService object, creating it from outside.